spec/support/mixins/pg_root_role_creator.rb

```diff
@@ -0,0 +1,37 @@
+require 'pg'
+require 'more_core_extensions/core_ext/hash'
+
+# This is simply to allow the dumps to be restored properly for the default
+# config that is usually in place on an appliance.
+#
+# Set up as a mixin so both CI and the local code can use this.  Lifted from
+# manageiq-appliance_console's internal database configuration.
+module PgRootRoleCreator
+  def create_root_role
+    with_connection do |conn|
+      pass = conn.escape_string("smartvm")
+      conn.exec("CREATE ROLE root WITH LOGIN CREATEDB SUPERUSER PASSWORD '#{pass}'")
+    end
+  end
+  module_function :create_root_role
+
+  def create_postgres_role
```

Comment:
Is this not always there by default?

Reply:
I don't think so.  I would have to research to give you a definitive answer, but my guess is why this was there in most cases was because the rpm for postgresql that we use probably also creates a postgres user, and thus, as part of it's `initdb` step, automatically creates the `postgres` role.

Locally, in my case, I end up with a `nicklamuro` role by default, since that is the user that I am running the specs as, so when the dump file gets to these lines:

```SQL
REVOKE ALL ON SCHEMA public FROM PUBLIC;
REVOKE ALL ON SCHEMA public FROM postgres;
GRANT ALL ON SCHEMA public TO postgres;
GRANT ALL ON SCHEMA public TO PUBLIC;
```

It falls over and dies.